src/static/js/AttributeManager.js

```diff
@@ -153,37 +153,36 @@ AttributeManager.prototype = _(AttributeManager.prototype).extend({
     return this.applyChangeset(builder);
   },
   
-  /*
-     Removes a specified attribute on a line
-     @param lineNum: the number of the affected line
-     @param attributeKey: the name of the attribute to remove, e.g. list
-
+ /**
+   * Removes a specified attribute on a line
+   *  @param lineNum the number of the affected line
+   *  @param attributeName the name of the attribute to remove, e.g. list
+   *  @param attributeValue if given only attributes with equal value will be removed
    */
    removeAttributeOnLine: function(lineNum, attributeName, attributeValue){
-     var loc = [0,0];
-     var builder = Changeset.builder(this.rep.lines.totalWidth());
-     var hasMarker = this.lineHasMarker(lineNum);
-     var attribs
-     var foundAttrib = false
-     
-     attribs = this.getAttributesOnLine(lineNum).map(function(attrib) {
-       if(attrib[0] === attributeName) {
-         foundAttrib = true
-         return [attributeName, null] // remove this attrib from the linemarker
-       }
-       return attrib
-     })
+       var builder = Changeset.builder(this.rep.lines.totalWidth());
+       var hasMarker = this.lineHasMarker(lineNum);
+       var found = false;
 
-     if(!foundAttrib) {
-       return
-     }
+       var attribs = _(this.getAttributesOnLine(lineNum)).map(function (attrib) {
+           if (attrib[0] === attributeName && (!attributeValue || attrib[0] === attributeValue)){
+               found = true;
+               return [attributeName, ''];
+           }
+           return attrib;
+       });
 
-     if(hasMarker){
-       ChangesetUtils.buildKeepRange(this.rep, builder, loc, (loc = [lineNum, 0]));
-       // If length == 4, there's [author, lmkr, insertorder, + the attrib being removed] thus we can remove the marker entirely
-       if(attribs.length <= 4) ChangesetUtils.buildRemoveRange(this.rep, builder, loc, (loc = [lineNum, 1]))
-       else ChangesetUtils.buildKeepRange(this.rep, builder, loc, (loc = [lineNum, 1]), attribs, this.rep.apool);
-     }
+       if (!found) {
+           return;
+       }
+       ChangesetUtils.buildKeepToStartOfRange(this.rep, builder, [lineNum, 0]);
+       var list = _.chain(attribs).filter(function(a){return !!a[1];}).map(function(a){return a[0];}).value();
+       //if we have marker and any of attributes don't need to have marker. we need delete it
+       if(hasMarker && !_.intersection(lineAttributes,list)){
```

Comment:
Also, what If the line has no marker?, then you apply attribs to the first char of the line. That may be ok, as there'd probably be no attribs in there anyway (since there's no line marker), but it's misleading at least. The assumption for putting the `if(hasMarker)` in there was that there are no attributes to be removed from the line if there's no line marker.